### DISCUSSION TITLE
Is it possible success and error handling in refetch

### DISCUSSION THREAD

**Discussion by gopusrikanth04** [TanStack/query#1687] [Q&A]
Hello Team.

I am facing problem while using  refetch . i need this type of execution order any help.

```
 const {status, data, refetch} =  useQuery('key', fetchData, { refetchOnWindowFocus: false  ,enabled:false});

const getData = () = >{

      //some code  to be executed

 refetch.then(()=>{
   if(success){
    // do something with the response
 }
 else{
      //  display  error messages
   }
  })
  
 //after refetch remain code to be executed
};

<button onClick={getData}>Get Posts</button>
```

**Comment by marinvirdol** [comment]
if you want an error to be thrown you should pass the `throwOnError: true` option to the `refetch` function.

Although in your use case I would do something like this
```
const [show, setShow] = useState(false)

const {status, data} = useQuery('key', () => fetchData(), {enabled: show})

...

<button onClick={() => setShow(true)}>Get Posts</button>
...
{show ? {/* display posts */} : null}
```

**Reply by TkDodo** [reply]
the docs for the `refetch` function say:

> If the query errors, the error will only be logged. If you want an error to be thrown, pass the throwOnError: true option

but as @marinvirdol already hinted, I would also use some local state and utilize the `enabled` prop rather than calling `refetch` imperatively.
